core/capabilities/ccip/oraclecreator/plugin.go

```diff
@@ -491,25 +514,36 @@ func createChainWriter(
 	execBatchGasLimit uint64,
 	chainFamily string,
 ) (types.ContractWriter, error) {
-	var fromAddress common.Address
+	var err error
+	var chainWriterConfig []byte
 	transmitter, ok := transmitters[types.NewRelayID(chainFamily, chainID)]
-	if ok {
-		// TODO: remove EVM-specific stuff
-		fromAddress = common.HexToAddress(transmitter[0])
-	}
-
-	chainWriterRawConfig, err := evmconfig.ChainWriterConfigRaw(
-		fromAddress,
-		defaultCommitGasLimit,
-		execBatchGasLimit,
-	)
-	if err != nil {
-		return nil, fmt.Errorf("failed to create chain writer config: %w", err)
-	}
 
-	chainWriterConfig, err := json.Marshal(chainWriterRawConfig)
-	if err != nil {
-		return nil, fmt.Errorf("failed to marshal chain writer config: %w", err)
+	switch chainFamily {
+	case relay.NetworkSolana:
+		var solConfig chainwriter.ChainWriterConfig
+		if solConfig, err = getSolanaChainWriterConfig(transmitter[0]); err == nil {
```

Comment:
Looking at how this transmitter map is [initialized](https://github.com/smartcontractkit/chainlink/blob/2450fff71db772d7e771babb5cbe1a55f5a51f84/core/capabilities/ccip/delegate.go#L288), the addresses are still EVM-specific. I assume CCIP team will update it, right?

Reply:
Good point, that too needs to be changed to support solana keystore too.
I think we need to do it.
It seems similar to the previous PR @archseer sent to support solana keystore.

We need a switch statement there per chain family, and for solana, do this:
solKeys, err := d.keystore.Solana().GetAll()

Plz check with @archseer. 1 thing I see missing is the solana chain-id here.

Reply:
Switch statement added, I used `d.keystore.Solana().GetAll()`